examples/main/java/io/milvus/resourcegroup/ResourceGroupManagement.java

```diff
@@ -0,0 +1,281 @@
+import io.milvus.param.collection.ShowCollectionsParam;
+import io.milvus.param.control.GetReplicasParam;
+import io.milvus.param.resourcegroup.CreateResourceGroupParam;
+import io.milvus.param.resourcegroup.DescribeResourceGroupParam;
+import io.milvus.param.resourcegroup.DropResourceGroupParam;
+import io.milvus.param.resourcegroup.ListResourceGroupsParam;
+import io.milvus.param.resourcegroup.TransferReplicaParam;
+import io.milvus.param.resourcegroup.UpdateResourceGroupsParam;
+import io.milvus.common.resourcegroup.ResourceGroupConfig;
+import io.milvus.common.resourcegroup.ResourceGroupLimit;
+import io.milvus.common.resourcegroup.ResourceGroupTransfer;
+import io.milvus.exception.MilvusException;
+import io.milvus.grpc.DescribeResourceGroupResponse;
+import io.milvus.grpc.GetLoadStateResponse;
+import io.milvus.grpc.GetReplicasResponse;
+import io.milvus.grpc.ListDatabasesResponse;
+import io.milvus.grpc.ListResourceGroupsResponse;
+import io.milvus.grpc.LoadState;
+import io.milvus.grpc.ShowCollectionsResponse;
+
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+public class ResourceGroupManagement {
+    public static String RECYCLE_RG = "__recycle_resource_group";
+    public static String DEFAULT_RG = "__default_resource_group";
+    static Integer RECYCLE_RG_REQUEST_NODE_NUM = 0;
+    static Integer RECYCLE_RG_LIMIT_NODE_NUM = 100000;
+
+    protected static final Logger logger = LoggerFactory.getLogger(ResourceGroupManagement.class);
+
+    private MilvusClient client;
+
+    public ResourceGroupManagement(MilvusClient client) {
+        this.client = client;
+    }
+
+    /**
+     * list all resource groups.
+     * 
+     * @return map of resource group name and resource group info.
+     */
+    public Map<String, ResourceGroupInfo> listResourceGroups() throws Exception {
+        // List all resource groups.
+        R<ListResourceGroupsResponse> response = client
+                .listResourceGroups(ListResourceGroupsParam.newBuilder().build());
+        ListResourceGroupsResponse resourceGroups = unwrap(response);
+
+        // Describe all resource groups.
+        Map<String, ResourceGroupInfo.Builder> result = new HashMap<>();
+        for (String resourceGroupName : resourceGroups.getResourceGroupsList()) {
+            R<DescribeResourceGroupResponse> resourceGroupInfoResp = client.describeResourceGroup(
+                    DescribeResourceGroupParam.newBuilder().withGroupName(resourceGroupName).build());
+            DescribeResourceGroupResponse resourceGroupInfo = unwrap(resourceGroupInfoResp);
+
+            ResourceGroupInfo.Builder builder = ResourceGroupInfo.newBuilder()
+                    .withResourceGroupName(resourceGroupName)
+                    .withRequestsNodeNum(resourceGroupInfo.getResourceGroup().getConfig().getRequests().getNodeNum())
+                    .withConfig(new ResourceGroupConfig(resourceGroupInfo.getResourceGroup().getConfig()));
```

Comment:
it's weird to pass rg config and requestNodeNum at same time

Reply:
redundant code here, I will remove it.